src/Nethermind/Nethermind.Synchronization/FastBlocks/SyncStatusList.cs

```diff
@@ -105,18 +105,21 @@ public bool TryGetInfosForBatch(int batchSize, Func<BlockInfo, bool> blockExist,
                 if (hasNonNull || !hasInserted)
                 {
                     CompileOutput(out infos);
+                    workingArray.Dispose();
                     return true;
                 }
 
                 // At this point, hasNonNull is false and hasInserted is true, meaning all entry in workingArray
                 // already exist. We switch to a bigger array to improve parallelization throughput
                 if (workingArray.Count < ParallelExistCheckSize)
                 {
+                    workingArray.Dispose();
```

Comment:
I see it's overwritten here. Still, manual disposal and reassignment would benefit from having a coarse grained using for the array, would it not?